packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1162,25 +1176,35 @@ export class Function extends FunctionBase {
    * function and undefined if not.
    */
   private getLoggingConfig(props: FunctionProps): CfnFunction.LoggingConfigProperty | undefined {
-    if ((props.applicationLogLevel || props.systemLogLevel) && props.logFormat !== LogFormat.JSON
-    && props.loggingFormat === undefined) {
+    if ((props.applicationLogLevel || props.applicationLogLevelV2 || props.systemLogLevel || props.systemLogLevelV2)
+      && props.logFormat !== LogFormat.JSON
+      && props.loggingFormat === undefined) {
       throw new Error(`To use ApplicationLogLevel and/or SystemLogLevel you must set LogFormat to '${LogFormat.JSON}', got '${props.logFormat}'.`);
     }
 
-    if ((props.applicationLogLevel || props.systemLogLevel) && props.loggingFormat !== LoggingFormat.JSON && props.logFormat === undefined) {
+    if ((props.applicationLogLevel || props.applicationLogLevelV2 || props.systemLogLevel || props.systemLogLevelV2)
+      && props.loggingFormat !== LoggingFormat.JSON && props.logFormat === undefined) {
       throw new Error(`To use ApplicationLogLevel and/or SystemLogLevel you must set LoggingFormat to '${LoggingFormat.JSON}', got '${props.loggingFormat}'.`);
     }
 
     if (props.logFormat && props.loggingFormat) {
       throw new Error('Only define LogFormat or LoggingFormat, not both.');
     }
```

Comment:
```suggestion
    if (props.logFormat && props.loggingFormat) {
      throw new Error('Only define LogFormat or LoggingFormat, not both.');
    }
    if ((props.applicationLogLevel || props.applicationLogLevelV2 || props.systemLogLevel || props.systemLogLevelV2)) {
          // we can assume loggingFormat or logFormat is undefined we already threw an error if both were defined
          if (props.logFormat !== LogFormat.JSON) {
              throw err ...
          }
          if (props.loggingFormat !== LoggingFormat.JSON){
              throw err ...
          }
    }
```
Need to double check if the tests still pass after this change but I think this should work